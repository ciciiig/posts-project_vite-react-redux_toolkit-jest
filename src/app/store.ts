import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit"
import counterReducer from "../features/counter/counterSlice"
import paginationReducer from "../features/pagination/paginationSlice"
import postsReducer from "../features/posts/postsSlice"

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    pagination: paginationReducer,
    posts: postsReducer,
  },
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
