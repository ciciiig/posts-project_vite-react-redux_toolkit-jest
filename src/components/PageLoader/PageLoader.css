.alertSkeleton-container {
    background-color: rgba(92, 60, 60, 0.88);
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    color: white;
    font-size: 46px;
}
