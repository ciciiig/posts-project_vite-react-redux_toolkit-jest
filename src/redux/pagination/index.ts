import { paginationSlice } from "./slice"
import { selectPagination } from "./selectors"
import {
  setCurrentPage,
  setPreviousPage,
  setNextPage,
  setMaxPages,
} from "./actions"
import { PaginationState } from "./type"

// export reducer
export const paginationReducer = paginationSlice.reducer
// export actions
export { setCurrentPage, setPreviousPage, setNextPage, setMaxPages }
// export selectors
export { selectPagination }
// export types
export type { PaginationState }
