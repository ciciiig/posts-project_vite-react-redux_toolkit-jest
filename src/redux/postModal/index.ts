import { postModalSlice } from "./slice"
import { selectPostModal } from "./selectors"
import {
  setClickedPostId,
  setIsOpen,
  setOriginalPost,
  setEditedPost,
} from "./actions"
import { PostModalState } from "./type"

// export reducer
export const postModalReducer = postModalSlice.reducer
// export selectors
export { selectPostModal }
// export actions
export { setClickedPostId, setIsOpen, setOriginalPost, setEditedPost }
// export types
export type { PostModalState }
