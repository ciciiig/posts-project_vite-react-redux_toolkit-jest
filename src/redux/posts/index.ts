import { postsSlice } from "./slice"
import { PostsState, Post } from "./type"
import { selectPosts } from "./selectors"
import {
  setCurrentPosts,
  setSearchValue,
  updatePost,
  setPatchRequest,
  fetchPosts,
  patchPost,
} from "./actions"

// export reducer
export const postsReducer = postsSlice.reducer
// export types
export type { PostsState, Post }
// export selectors
export { selectPosts }
// export actions
export {
  setCurrentPosts,
  setSearchValue,
  updatePost,
  setPatchRequest,
  fetchPosts,
  patchPost,
}
