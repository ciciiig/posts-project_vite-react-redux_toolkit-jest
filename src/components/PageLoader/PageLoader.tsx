import "./PageLoader.css"
import { FC } from "react"
import { useAppSelector } from "../../redux/hooks"
import { selectPosts } from "../../redux/posts"

export const PageLoader: FC = () => {
  const postState = useAppSelector(selectPosts)
  let skeletonAlertContent = null

  if (postState.skeletonStatus === "loading") {
    skeletonAlertContent = (
      <div className="alertSkeleton-container">Loading...</div>
    )
  }

  if (postState.skeletonStatus === "failed") {
    skeletonAlertContent = (
      <div className="alertSkeleton-container">
        <div>{postState.error ?? "Some error"}</div>
      </div>
    )
  }

  if (postState.skeletonStatus === "idle") {
    skeletonAlertContent = null
  }

  return skeletonAlertContent
}
