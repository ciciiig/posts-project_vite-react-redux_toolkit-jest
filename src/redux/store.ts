import { configureStore } from "@reduxjs/toolkit"
import { paginationReducer } from "./pagination"
import { postsReducer } from "./posts"
import { postModalReducer } from "./postModal"

export const store = configureStore({
  reducer: {
    pagination: paginationReducer,
    posts: postsReducer,
    singlePostWindow: postModalReducer,
  },
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
