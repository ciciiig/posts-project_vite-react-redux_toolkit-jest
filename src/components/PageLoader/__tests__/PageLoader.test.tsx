import { PageLoader } from "../PageLoader"
import { useAppSelector } from "../../../redux/hooks"
import { RootState } from "../../../redux/store"
import { clone } from "ramda"

import { render, screen } from "@testing-library/react"
import { current } from "@reduxjs/toolkit"

jest.mock("../../../redux/hooks", () => ({
  ...jest.requireActual("../../../redux/hooks"),
  useAppSelector: jest.fn(),
}))

describe("Test <PageLoader />", () => {
  const initialState = {
    posts: {
      skeletonStatus: "idle",
      error: undefined,
    },
  } as RootState
  let currentState: RootState

  beforeEach(() => {
    jest.clearAllMocks()

    currentState = clone(initialState)

    jest
      .mocked(useAppSelector)
      .mockImplementation((selectorFn) => selectorFn(currentState))
  })

  it("should  not render skeletonAlertContent when skeletonStatus is idle", () => {
    const { container } = render(<PageLoader />)
    expect(container).toMatchSnapshot()
  })
  it("should render skeletonAlertContent with loading when skeletonStatus is loading", () => {
    currentState.posts.skeletonStatus = "loading"

    const { container } = render(<PageLoader />)
    expect(container).toMatchSnapshot()
  })
  it("should render skeletonAlertContent with Error when skeletonStatus is failed", () => {
    currentState.posts.skeletonStatus = "failed"
    currentState.posts.error = "Some error should occure"

    const { container } = render(<PageLoader />)
    expect(container).toMatchSnapshot()
  })
})
