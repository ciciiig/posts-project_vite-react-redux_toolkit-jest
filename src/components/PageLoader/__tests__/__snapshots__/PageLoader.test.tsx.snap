// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PageLoader /> should  not render skeletonAlertContent when skeletonStatus is idle 1`] = `<div />`;

exports[`Test <PageLoader /> should render skeletonAlertContent with Error when skeletonStatus is failed 1`] = `
<div>
  <div
    class="alertSkeleton-container"
  >
    <div>
      Some error should occure
    </div>
  </div>
</div>
`;

exports[`Test <PageLoader /> should render skeletonAlertContent with loading when skeletonStatus is loading 1`] = `
<div>
  <div
    class="alertSkeleton-container"
  >
    Loading...
  </div>
</div>
`;
